src/gluonts/dataset/repository/_zenodo.py

```diff
@@ -0,0 +1,195 @@
+        block_size  : int, optional
+            Size of each block (in tqdm units) [default: 1].
+        tsize  : int, optional
+            Total size (in tqdm units). If [default: None] remains unchanged.
+        """
+        if tsize is not None:
+            tqdm.total = tsize
+        tqdm.update((block - last_b[0]) * block_size)
+        last_b[0] = block
+
+    return update_to
+
+
+def download_dataset(description: Dict[str, str], path: Path):
+    file = description["file"]
+    with tqdm(
+        [],
+        unit="B",
+        unit_scale=True,
+        unit_divisor=1024,
+        miniters=5,
+        desc=f"download {file}",
+    ) as _tqdm:
+        request.urlretrieve(
+            f"{ROOT}/{description['record']}/files/{file}",
+            filename=str(path / description["file"]),
+            reporthook=urllib_retrieve_hook(_tqdm),
+        )
+
+
+def frequency_converter(freq: str):
+    parts = freq.split("_")
+    if len(parts) == 1:
+        return freq[0].upper()
+    if len(parts) == 2 and parts[0].isnumeric():
+        return f"{parts[0]}{parts[1].upper()}"
+
+
+def save_metadata(
+    dataset_path: Path, cardinality: int, freq: str, prediction_length: int
+):
+    with open(dataset_path / "metadata.json", "w") as file:
+        json.dump(
+            metadata(
+                cardinality=cardinality,
+                freq=freq,
+                prediction_length=prediction_length,
+            ),
+            file,
+        )
+
+
+def save_datasets(path: Path, data: List[Dict], train_offset: int):
+    train = path / "train"
+    test = path / "test"
+
+    train.mkdir(exist_ok=True)
+    test.mkdir(exist_ok=True)
+
+    with open(train / "data.json", "w") as train_fp, open(
+        test / "data.json", "w"
+    ) as test_fp:
+        for data_entry in tqdm(
+            data, total=len(data), desc="creating json files"
+        ):
+            dic = to_dict(
+                target_values=data_entry["target"],
+                start=str(data_entry["start_timestamp"]),
+            )
+
+            test_fp.write(json.dumps(dic))
+            test_fp.write("\n")
+
+            dic["target"] = dic["target"][:-train_offset]
+            train_fp.write(json.dumps(dic))
+            train_fp.write("\n")
+
+
+def clean_up_dataset(dataset_path: Path, file_names: List[str]):
+    for file in file_names:
+        file_path = dataset_path / file
+        file_path.unlink()
+
+
+def generate_forecasting_dataset(dataset_path: Path, dataset_name: str):
+    ds_info = dataset_info[dataset_name]
+    dataset_path.mkdir(exist_ok=True)
+
+    file = ds_info["file"]
+    file_path = dataset_path / file
+    download_dataset(ds_info, dataset_path)
+    with ZipFile(file_path, "r") as zip:
+        file_names = zip.namelist()
+        # TODO better exception text
+        assert len(file_names) == 1, "Too many files"
+        for member in tqdm(zip.infolist(), desc=f"Extracting {file}"):
```

Comment:
Why use `infolist()` here, when you have used `namelist()` before?

Reply:
Removed that all together, using now `extractall`